PORTF-916 u boot add command for resetting ethernity

yba-siklu opened the pull request:
## Description
Added Ethernity cli reset command.

This PR also has a new version of cpld_reg.h for the EH8020 verified against "CPLD Register Description.xlsx".

## Test
 * None, still in debugging mode

## References
JIRA:PORTF-1082, PORTF-916

yba-siklu committed:
Added a README for Siklu's changes to U-Boot

This readme helps you locate the files that Siklu changed.

baruchsiach commented on board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c:
```diff
@@ -0,0 +1,41 @@
+	int rc = 0;
+	T_CPLD_LOGIC_WD_RW_REGS reset_reg;
+
+	reset_reg.s.cfg_fpga_8020_rst = 1;
```
What about all other bits in `uint8`. They are left uninitialized here. I guess the CPLD currently ignores these bits. But a future CPLD might behave strangely when random bits are set.

yba-siklu replied:
Fixed. Thanks.

baruchsiach commented on board/freescale/mx6ullevk/cpld_reg.h:
```diff
@@ -123,46 +144,47 @@ typedef union
-		u8 cfg_rst_n:1;		// When asserted to '0' resets the Ethernity switch.
-	} s;
-} T_CPLD_LOGIC_ETHERNITY_REGS;
-
```
This removes the definition added in the previous commit. I would be nicer to squash this change into the previous commit.

yba-siklu replied:
Fixed, thanks.

yba-siklu committed:
PORTF-916 Ethernity reset command "ethernity_rst"

This command has no parameters.

yba-siklu committed:
PORTF-1082 New cpld_reg.h for EH8020

Verified against "CPLD Register Description.xlsx",
06/06/2022, 07:34:14, by Moshe Levy

baruchsiach commented on board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c:
```diff
@@ -0,0 +1,41 @@
+	rc = siklu_cpld_write(R_CPLD_LOGIC_WD_RW, reset_reg.uint8);
+	if (rc != 0) {
+		printf("%s() ERROR writing to CPLD, line %d\n", __func__, __LINE__);
+		return rc;
```
Nitpick: this line is redundant.

yba-siklu replied:
Fixed.
